pkg/e2ecluster/e2ehelpers/curl.go

```diff
@@ -0,0 +1,107 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package e2ehelpers
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+)
+
+const (
+	defaultConnectTimeout = 5 * time.Second
+	defaultMaxTime        = 20 * time.Second
+)
+
+type CurlResultFormat string
+
+const (
+	CurlResultFormatStats    CurlResultFormat = `time-> DNS: '%{time_namelookup}(%{remote_ip})', Connect: '%{time_connect}', Transfer '%{time_starttransfer}', total '%{time_total}'`
+	CurlResultFormatHTTPCode CurlResultFormat = "%%{http_code}"
+)
+
+type CurlOpts struct {
```

Comment:
Overall style question: This this struct need to be public? I noticed it is public in other helpers too, but I don't think it needs to be.

Reply:
Good find, I don't think it needs to be public. I'll change this and add another commit to change existing `*Opts` structs.